src/source-napatech.c

```diff
@@ -157,6 +172,315 @@ void TmModuleNapatechDecodeRegister(void)
+    }
+
+    struct IPv4Tuple4 v4Tuple;
+    struct IPv6Tuple4 v6Tuple;
+
+    switch (layer3) {
+        case RTE_PTYPE_L3_IPV4:
+        {
+            struct ipv4_hdr *pIPv4_hdr = (struct ipv4_hdr *) (packet + ntpv->dyn3->offset0);
+            v4Tuple.da = pIPv4_hdr->dst_addr;
+            v4Tuple.sa = pIPv4_hdr->src_addr;
+            break;
+        }
+        case RTE_PTYPE_L3_IPV6:
+        {
+            struct ipv6_hdr *pIPv6_hdr = (struct ipv6_hdr *) (packet + ntpv->dyn3->offset0);
+            memcpy(&(v6Tuple.da), pIPv6_hdr->dst_addr, 16);
+            memcpy(&(v6Tuple.sa), pIPv6_hdr->src_addr, 16);
+            break;
+        }
+        default:
+        {
+            return 0;
+        }
+    }
+
+    switch (layer4) {
+        case RTE_PTYPE_L4_TCP:
+        {
+            struct tcp_hdr *tcp_hdr = (struct tcp_hdr *) (packet + ntpv->dyn3->offset1);
+
+            if (layer3 == RTE_PTYPE_L3_IPV4) {
+                v4Tuple.dp = tcp_hdr->dst_port;
+                v4Tuple.sp = tcp_hdr->src_port;
+                memcpy(&(flow_match.keyData), &v4Tuple, sizeof(v4Tuple));
+                flow_match.keyId = NAPATECH_KEYTYPE_IPV4;
+            } else {
+                v6Tuple.dp = tcp_hdr->dst_port;
+                v6Tuple.sp = tcp_hdr->src_port;
+                memcpy(&(flow_match.keyData), &v6Tuple, sizeof(v6Tuple));
+                flow_match.keyId = NAPATECH_KEYTYPE_IPV6;
+            }
+            flow_match.ipProtocolField = 6;
+            break;
+        }
+        case RTE_PTYPE_L4_UDP:
+        {
+            struct udp_hdr *udp_hdr = (struct udp_hdr *) (packet + ntpv->dyn3->offset1);
+
+            if (layer3 == RTE_PTYPE_L3_IPV4) {
+                v4Tuple.dp = udp_hdr->dst_port;
+                v4Tuple.sp = udp_hdr->src_port;
+                flow_match.keyId = NAPATECH_KEYTYPE_IPV4;
+                memcpy(&(flow_match.keyData), &v4Tuple, sizeof(v4Tuple));
+            } else {
+                v6Tuple.dp = udp_hdr->dst_port;
+                v6Tuple.sp = udp_hdr->src_port;
+                flow_match.keyId = NAPATECH_KEYTYPE_IPV6;
+                memcpy(&(flow_match.keyData), &v6Tuple, sizeof(v6Tuple));
+            }
+            flow_match.ipProtocolField = 17;
+            break;
+        }
+        default:
+        {
+            return 0;
+        }
+    }
+
+    flow_match.op = 1;  /* program flow */
+    flow_match.gfi = 0; /* Do not generate FlowInfo records */
+    flow_match.tau = 1; /* tcp automatic unlearn */
+
+    if (PACKET_TEST_ACTION(p, ACTION_DROP)) {
+        flow_match.keySetId = NAPATECH_FLOWTYPE_DROP;
+    } else {
+        if (is_inline) {
+            flow_match.keySetId = NAPATECH_FLOWTYPE_PASS;
+        } else {
+            flow_match.keySetId = NAPATECH_FLOWTYPE_DROP;
+        }
+    }
+
+    status = NT_FlowWrite(*phFlowStream, &flow_match, 200);
+    if (status == NT_STATUS_TIMEOUT) {
+        SCLogInfo("NT_FlowWrite returned NT_STATUS_TIMEOUT");
```

Comment:
Is a timeout considered success or failure?

Reply:
Added comment and a check for a more serious error.